theia-full-docker/Dockerfile

```diff
@@ -220,40 +321,60 @@ ENV PATH $DART_SDK/bin:/theia/.pub-cache/bin:$PATH
 ## User account
 RUN adduser --disabled-password --gecos '' theia && \
     adduser theia sudo && \
-    echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers;
+    echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers
+ADD settings.json /home/theia/.theia/
 
 RUN chmod g+rw /home && \
     mkdir -p /home/project && \
     mkdir -p /home/theia/.pub-cache/bin && \
     mkdir -p /usr/local/cargo && \
     mkdir -p /usr/local/go && \
     mkdir -p /usr/local/go-packages && \
+    chown -R theia:theia /home/theia && \
+    chown -R theia:theia /home/project && \
     chown -R theia:theia /usr/local/cargo && \
     chown -R theia:theia /usr/local/go && \
-    chown -R theia:theia /usr/local/go-packages && \
-    chown -R theia:theia /home/theia && \
-    chown -R theia:theia /home/project;
+    chown -R theia:theia /usr/local/go-packages
+
 
-#Theia
+# Theia application
 ##Needed for node-gyp, nsfw build
-RUN apt-get update && apt-get install -y python build-essential
+RUN apt-get update && apt-get install -y python build-essential && \
+  apt-get clean && \
+  apt-get autoremove -y && \
+  rm -rf /var/cache/apt/* && \
+  rm -rf /var/lib/apt/lists/* && \
+  rm -rf /tmp/*
 
 USER theia
-ARG version=latest
 WORKDIR /home/theia
 ADD $version.package.json ./package.json
-ARG GITHUB_TOKEN
-# using "NODE_OPTIONS=..." to avoid out-of-memory problem in CI
-RUN yarn --cache-folder ./ycache && rm -rf ./ycache && \
-    NODE_OPTIONS="--max_old_space_size=4096" yarn theia build ;\
-    yarn theia download:plugins
+
+RUN if [ "$strip" = "true" ]; then \
+yarn --pure-lockfile && \
+    NODE_OPTIONS="--max_old_space_size=4096" yarn theia build && \
+    yarn theia download:plugins && \
+    yarn --production && \
+    yarn autoclean --init && \
+    echo *.ts >> .yarnclean && \
+    echo *.ts.map >> .yarnclean && \
+    echo *.spec.* >> .yarnclean && \
+    yarn autoclean --force && \
+    yarn cache clean \
+;else \
+yarn --cache-folder ./ycache && rm -rf ./ycache && \
+     NODE_OPTIONS="--max_old_space_size=4096" yarn theia build && yarn theia download:plugins \
+;fi
+
 EXPOSE 3000
+# configure Theia
 ENV SHELL=/bin/bash \
-    THEIA_DEFAULT_PLUGINS=local-dir:/home/theia/plugins
+    THEIA_DEFAULT_PLUGINS=local-dir:/home/theia/plugins  \
+    # configure user Go path
+    GOPATH=/home/project
 
-ENV GOPATH=/home/project
-ENV PATH=$PATH:$GOPATH/bin
 ENV LC_ALL=C.UTF-8
 ENV LANG=C.UTF-8
 
-ENTRYPOINT [ "yarn", "theia", "start", "/home/project", "--hostname=0.0.0.0" ]
+ENTRYPOINT [ "node", "/home/theia/src-gen/backend/main.js", "/home/project", "--hostname=0.0.0.0" ]
+# ENTRYPOINT [ "yarn", "theia", "start", "/home/project", "--hostname=0.0.0.0" ]
```

Comment:
Remove commented-out line?

Reply:
BTW, I am curious: did you find a functional difference between the two forms of `ENTRYPOINT` above?

Reply:
I'm going to delete the commented line after you review and tests are finished, maybe there will be some more changes. I didn't noticed any difference between the two lines, just took the change out of the cpp docker script.